lib/spack/spack/spec.py

```diff
@@ -1825,6 +1829,73 @@ def process_hash_bit_prefix(self, bits):
         """Get the first <bits> bits of the DAG hash as an integer type."""
         return spack.util.hash.base32_prefix_bits(self.process_hash(), bits)
 
+    def _lookup_hash(self):
+        """Lookup just one spec with an abstract hash, returning a spec from the the environment,
+        store, or finally, binary caches."""
+        import spack.environment
```

Comment:
Yep agree with this. I think having some central place for managing the world of known Specs would be useful.

Reply:
I like this idea for improvement, but I'm having trouble thinking where else this information would belong, it's not really a natural fit anywhere. It could live in `spack.spec` but outside the `Spec` class, but that doesn't feel satisfying really.